app/client/ts/signIn-utils.ts

```diff
@@ -0,0 +1,38 @@
+import { safeClient } from "./util";
+
+//load the current longNames into the list
+async function loadNames(): Promise<void> {
+    const dropdown : HTMLSelectElement = <HTMLSelectElement> this.refs.nameDropdown;
+
+    const length : number = dropdown.length;
+
+    for (let i = 0; i < length; i++) {
+        dropdown.remove(0);
+    }
+
+    await safeClient.dns.getLongNames().then((data) => {
+        data.forEach((name) => { dropdown.add(new Option(name, name)); });
+    });
+}
+
+export async function addLongName(): Promise<void> {
+    const inputField : HTMLInputElement = <HTMLInputElement> this.refs.longName;
+    const name: string = inputField.value;
+    if (name) {
+        await safeClient.dns.register(name).then(() => {
+            const errorMsg : HTMLElement = <HTMLElement> this.refs.error_msg;
+            errorMsg.innerHTML = `LongName ${name} successfully registered!`;
+        }, (err) => {
+            const errorMsg : HTMLElement = <HTMLElement> this.refs.error_msg;
+            errorMsg.innerHTML = "There was an error processing your request. Please choose a different LongName.";
```

Comment:
Maybe in the future we can report some reason back to the user about why their LongName was rejected? e.g. it was too long or didn't meet <x> rule

(Doesn't need to be this iteration.)